packages/lime-proto-babeld/files/usr/lib/lua/lime/proto/babeld.lua

```diff
@@ -103,17 +103,54 @@ function babeld.setup_interface(ifname, args)
 	local vlanProto = args[3] or "8021ad"
 	local nameSuffix = args[4] or "_babeld"
 
+	local addIPtoIf = true
+
+	--! If Babeld is without VLAN (vlanId is 0) it cannot run directly
+	--! on ethernet interfaces which are inside of a bridge (e.g. eth0 or eth0.1)
+	--! because they cannot have an IPv6 Link-Local, so Babeld has to run on 
+	--! the bridge interface br-lan
+	--! If Babeld's Hello packets run over Batman-adv (whose bat0 is also 
+	--! included in br-lan), the links will have a wrong quality metric,
+	--! so these hello on bat0 have to be filtered
+	if tonumber(vlanId) == 0 then
+		local hasBatman = false
+		local babeldOverBatman = config.get_bool("network", "babeld_over_batman")
+		local hasLan = false
+		for _,protoArgs in pairs(config.get("network", "protocols")) do
+			local proto =  utils.split(protoArgs, network.protoParamsSeparator)[1]
+			if(proto == "lan") then hasLan = true
+			elseif(proto == "batadv") then hasBatman = true end
+		end
+	
+		if hasLan and ifname:match("^eth%d") then
```

Comment:
What happens with wan port? Or port manually configured with ground routing?

Reply:
Also consider to move this to configure method as it don't seems to need to be run for each interface, run it once seems enough

Reply:
> What happens with wan port?

Woops... With this PR, WAN port is not anymore used by Babeld.
This was unintentional, sorry.
Anyway WAN port is used for connecting to non-LibreMesh networks, so Babeld is not needed there.
I am ok with fixing it if we want Babeld on WAN.
Rather than checking for lan protocol, I can check for the presence of the interface into the 

```
config interface 'lan'
	list ifname 'bat0'
	list ifname 'eth0.1'
```
table.

> Or port manually configured with ground routing?

Ports configured with specific configuration are not treated by the whole section.
Do you mean that even if ground routing ports have a specific configuration, they should trigger this fix anyway? Are them included into br-lan bridge?

> Also consider to move this to configure method as it don't seems to need to be run for each interface, run it once seems enough

Sounds good!
Currently it runs more than once even if running once would be ok.
Running more than once is not detrimental but I agree that it is a waste.
If I put this into the configure, it would be executed even if no ethernet lan port is present, but I suppose this is ok, right?

Reply:
>>   Also consider to move this to configure method as it don't seems to need to be run for each interface, run it once seems enough

> If I put this into the configure, it would be executed even if no ethernet lan port is present, but I suppose this is ok, right?

Wait, the ebtables rule can be in the configuration part, but the `ifname = "br-lan"` stuff has to stay into the setup interface one...
This is a mess.

Reply:
>Ports configured with specific configuration are not treated by the whole section.
> Do you mean that even if ground routing ports have a specific configuration, they should trigger this fix anyway? Are them included into br-lan bridge?

It all depends on what kind of configuration they got, if they have babled specified as proto they get there too

Reply:
> This is a mess.

Reply:
> > Ports configured with specific configuration are not treated by the whole section.

I was wrong, interfaces with specific configuration are treated by this section.

> > Do you mean that even if ground routing ports have a specific configuration, they should trigger this fix anyway? Are them included into br-lan bridge?

The answer to my question was: no, interfaces configured via ground routing should not be touched by this fix and they will never be included in br-lan.

> It all depends on what kind of configuration they got, if they have babled specified as proto they get there too

Interfaces managed by ground routing will never have VLAN ID 0, right? If this is correct, they will never trigger the fix and the problem is solved. Otherwise, how can I check if the interface is also configured for ground routing?